nilearn/glm/second_level/second_level.py

```diff
@@ -918,7 +925,9 @@ def non_parametric_inference(
     """
     _check_second_level_input(second_level_input, design_matrix)
     _check_confounds(confounds)
-    _check_design_matrix(design_matrix)
+    if not isinstance(design_matrix, (list, tuple)):
+        design_matrix = [design_matrix]
```

Comment:
Design matrix is not expected to be a list or tuple in docstring. I think this part could also be implemented as lines 498-508.

Reply:
the docstring should be upated then.

Reply:
I think I was doing this because check_and_load_tables expects list or tuple: I could probably update it rather than change the public API

Reply:
As far as I understand, previously design_matrix could only be pandas.DataFrame. But now we allow it to be CSV or TSV so that it can be loaded from a file. It should not be a list or tuple. 

I think we can just delete line 928.